cmd/log/log.go

```diff
@@ -1,9 +1,18 @@
 package log
 
-import "github.com/openshift/hypershift/cmd/util"
+import (
+	"go.uber.org/zap/zapcore"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+)
 
-var log = util.Log
+var log = zap.New(zap.UseDevMode(true), func(o *zap.Options) {
+	o.TimeEncoder = zapcore.RFC3339TimeEncoder
+})
 
 func Error(err error, msg string, keysAndValues ...interface{}) {
 	log.Error(err, msg, keysAndValues...)
 }
+
+func Info(msg string, keysAndValues ...interface{}) {
+	log.Info(msg, keysAndValues...)
+}
```

Comment:
What's the reason `log` is private and the methods are re-exported from this package? Why not make `log` public and delete the `Error` and `Info` functions?

Reply:
I see to reasons for that:
1. This way we can call the logger with `log.Info(...)` instead of `log.Log.Info(...)`
2. This will allow future modifications for the logger, without changing all the callers (change only inside the wrapper functions)

Reply:
> I see two reasons for that:
> 
> 1. This way we can call the logger with `log.Info(...)` instead of `log.Log.Info(...)`

This can already be effectively achieved if `Log` is public. Users can specify a dot import alias:

```go
import (
  . "github.com/openshift/hypershift/cmd/log"
)

func foo() {
  Log.Info(...)
}
```

The user could also alias it to whatever package name they want.

> 2. This will allow future modifications for the logger, without changing all the callers (change only inside the wrapper functions)

What you're describing sounds like a logging interface. There's already a history of competing logging interfaces for Go, and we're currently not using any of them. Instead, we're just linking directly to Zap. If we wanted to switch to an interface, I think we need design consensus from the team, and we would first probably want to investigate using [https://github.com/go-logr/logr](logr), which is the interface controller-runtime is using (and we're tightly coupled to controller-runtime). But part of the reason to link directly to Zap is to take advantage of its specific features instead of using a lowest common denominator interface, so there's definitely a discussion to be had.

Reply:
Ok, I will change it now
This will require to change all lines using the logger from `log.` to `Log.`

Reply:
The `make verify` doesn't allow dot imports:
`cmd/bastion/aws/create.go:13:2: should not use dot imports (ST1001)`

Reply:
The linter isn't always the final arbiter of style here, have to get more feedback from the rest of the team. Another option would be for users to declare a local logger, like:

```go
import (
  logger "github.com/openshift/hypershift/cmd/log"
)

var log = logger.Log
```

But I can't speak for everyone on the team. This is a really subjective.

Reply:
I'm not a fan of underscore imports either, because the compiler can't detect when they become unused. Both using `log.Log`  or declaring a package-level variable like in your comment above seems fine to me.

Reply:
Declaring a package-level variable is actually how it was before this PR
In each package defined `log.go` file and defined the package variable in it
Do you prefer to keep the code this way?

Reply:
Either that or changing the log calls to `log.Log.${Method}`, both are fine with me

Reply:
I changed the code in this PR call: `log.Log.${Method}`
In case you prefer to use the package-level variable we can keep the previous implementation and close this PR